apps/api/src/domain/models/media-item/commands/publish-media-item.command-handler.ts

```diff
@@ -0,0 +1,57 @@
+import { CommandHandler } from '@coscrad/commands';
+import { Inject } from '@nestjs/common';
+import { InternalError } from '../../../../lib/errors/InternalError';
+import { isNotFound } from '../../../../lib/types/not-found';
+import { RepositoryProvider } from '../../../../persistence/repositories/repository.provider';
+import { ResultOrError } from '../../../../types/ResultOrError';
+import { Valid } from '../../../domainModelValidators/Valid';
+import { IIdManager } from '../../../interfaces/id-manager.interface';
+import { InMemorySnapshot, ResourceType } from '../../../types/ResourceType';
+import buildInMemorySnapshot from '../../../utilities/buildInMemorySnapshot';
+import { BaseUpdateCommandHandler } from '../../shared/command-handlers/base-update-command-handler';
+import ResourceNotFoundError from '../../shared/common-command-errors/ResourceNotFoundError';
+import { IEvent } from '../../shared/events/interfaces/event.interface';
+import { MediaItem } from '../entities/media-item.entity';
+import { MediaItemPublished } from './media-item-published.event';
+import { PublishMediaItem } from './publish-media-item.command';
+
+@CommandHandler(PublishMediaItem)
+export class PublishMediaItemCommandHandler extends BaseUpdateCommandHandler<MediaItem> {
+    constructor(
+        protected readonly repositoryProvider: RepositoryProvider,
+        @Inject('ID_MANAGER') protected readonly idManager: IIdManager,
+        protected readonly resourceType: ResourceType
+    ) {
+        super(repositoryProvider, idManager, ResourceType.mediaItem);
```

Comment:
It's really too bad that we have to flow dependencies through like this. We aren't fully leveraging Nest's DI capabilities.

Reply:
[here](https://www.pivotaltracker.com/story/show/182602926) is a related story.